cmd/podmanV2/registry/config.go

```diff
@@ -0,0 +1,57 @@
+package registry
+
+import (
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	RootRequired = "RootRequired"
+)
+
+var (
+	PodmanOptions entities.PodmanConfig
+)
+
+// NewPodmanConfig creates a PodmanConfig from the environment
+func NewPodmanConfig() entities.PodmanConfig {
+	if err := libpod.SetXdgDirs(); err != nil {
+		logrus.Errorf(err.Error())
+		os.Exit(1)
+	}
+
+	var mode entities.EngineMode
+	switch runtime.GOOS {
+	case "darwin":
+		fallthrough
+	case "windows":
+		mode = entities.TunnelMode
+	case "linux":
+		mode = entities.ABIMode
+	default:
+		logrus.Errorf("%s is not a supported OS", runtime.GOOS)
+		os.Exit(1)
+	}
+
+	// TODO: Is there a Cobra way to "peek" at os.Args?
+	for _, v := range os.Args {
+		if strings.HasPrefix(v, "--remote") {
```

Comment:
This doesn't seem safe - what if it's `--remote=false`? Why can't we pass the Cobra CLI object in here?

Reply:
This is before the cobra parse has happened.  In the next PR, I'll beef up the check.